api/v1alpha1/hypershiftlogforwarder_types.go

```diff
@@ -0,0 +1,57 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	loggingv1 "github.com/openshift/cluster-logging-operator/apis/logging/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// HyperShiftLogForwarderSpec defines the desired state of HyperShiftLogForwarder
+type HyperShiftLogForwarderSpec struct {
```

Comment:
Does this have to be an explicitly named type in lieu of straight composition?  Can it not be something like:
```
type HyperShiftLogForwarderSpec struct {
	loggingv1.ClusterLogForwarderSpec
}
```

Alternatively, I'm wondering if you could simply alias the type too to achieve the reuse.  cc @alanconway

Reply:
@bmeng It can be composed like Jeff suggests, and that is probably a good idea. The JSON tag you need is:
```
// HyperShiftLogForwarderSpec defines the desired state of HyperShiftLogForwarder
type HyperShiftLogForwarderSpec struct {
	loggingv1.ClusterLogForwarderSpec `json:",inline"`
}
```
That will JSON marshal exactly as ClusterLogForwarderSpec, but is a distinct type.

@jcantrill I made it a separate type because the HLF is watched by a different operator, from a different cluster, that is potentially of a different version to the local CLF. 

I originally wanted the HLO to watch the managed CLF directly (no need for extra HLF resource) and just extract the api-audit relevant parts -  would have been a better user experience. However the risk of version skew between  hosted and management clusters is what made me feel safer with a totally separate resource.